src/System.Private.CoreLib/shared/System/Exception.cs

```diff
@@ -164,6 +161,21 @@ public override string ToString()
             return s;
         }
 
+        // Returns the indented inner exception's ToString() prefixed with a newline.
+        // If there is no inner exception, returns empty string.
+        protected internal virtual string InnerExceptionToString(Exception? inner = null)
+        {
+            inner ??= _innerException;
+            if (inner == null)
+                return "";
+
+            string s = inner.ToString();
+            string[] arr = s.Split(new[] { "\r\n", "\n" }, StringSplitOptions.None);
+            s = String.Join(Environment.NewLine + "      ", arr);
```

Comment:
This is adding a lot of garbage allocation.  I realize this is on exceptional paths, but it seems unnecessarily expensive for the value it's providing, IMO.

Reply:
Sure, I'll change to do this manually.

Reply:
I changed to StringBuilder - Replace seems not too inefficient.